app/decorators/application_decorator.rb

```diff
@@ -21,12 +21,27 @@ def status_colour
   end
 
   def status_tags
+    object.has_vacancies? ? status_tags_for_vacancies : status_tags_for_no_vacancies
+  end
+
+  def status_tags_for_vacancies
+    {
+      published: { text: "Open", colour: "turquoise" },
+      withdrawn: { text: "Withdrawn", colour: "red" },
+      empty: { text: "Empty", colour: "grey" },
+      draft: { text: "Draft", colour: "grey" },
+      published_with_unpublished_changes: { text: "Open&nbsp;*", colour: "turquoise" },
+      rolled_over: { text: "Rolled over", colour: "yellow" },
+    }
+  end
+
+  def status_tags_for_no_vacancies
```

Comment:
The `status_tags_for_vacancies` and its counterpart can both be private as they're not publicly accessible. Also, `status_tags_for_no_vacancies` could just re-use the hash returned from `status_tags_for_vacancies` and merge a new one with the keys to override without having to duplicate most of it

Reply:
oh cool, I didn't realise you could do that 😅 

Done [here](https://github.com/DFE-Digital/publish-teacher-training/pull/2900/commits/d11bdaf30036ea38f83cffd21ae52ae39cb7e66d)

Reply:
That is what we're after, right?

Reply:
Yeah something like that :)